src/Arcus.Testing.Tests.Unit/Assert_/AssertCsvTests.cs

```diff
@@ -240,6 +240,61 @@ public void CompareWithIgnoredRowAndColumnOrder_WithShuffledCsv_Succeeds()
             });
         }
 
+        [Property]
+        public void CompareWithIgnoredColumnOrderAndColumnIndexes_WithShuffledCsv_FailsWithDescription()
+        {
+            // Arrange
+            TestCsv expected = TestCsv.Generate();
+            TestCsv actual = expected.Copy();
+            int[] possibleIndexes = Enumerable.Range(0, expected.ColumnCount).ToArray();
+            int ignoredIndex = Bogus.PickRandom(possibleIndexes);
+
+            actual.ShuffleColumns();
+
+            CompareShouldFailWithDescription(
+                expected,
+                actual,
+                options =>
+                {
+                    options.ColumnOrder = AssertCsvOrder.Ignore;
+                    options.IgnoreColumnIndex(ignoredIndex);
+                },
+                "cannot compare",
+                "column indexes can only be ignored when column order is included",
+                nameof(AssertCsvOptions.IgnoreColumnIndex), AssertCsvOrder.Ignore.ToString()
+            );
+        }
+
+        [Property]
+        public void CompareWithIgnoredColumnIndex_WithSameCsv_Succeeds()
+        {
+            // Arrange
+            TestCsv expected = TestCsv.Generate();
+            TestCsv actual = expected.Copy();
+            int[] possibleIndexes = Enumerable.Range(0, expected.ColumnCount).ToArray();
+            int ignoredIndex = Bogus.PickRandom(possibleIndexes);
```

Comment:
We have duplicated the random pick of a column index three times -> maybe this can even be a property of the `TestCsv`? Otherwise a method with would also suffice.

Reply:
Fixed in [66c3df0](https://github.com/arcus-azure/arcus.testing/pull/161/commits/66c3df02e8a59922ff59d4611e78663cdb04b6aa)